### PR TITLE
NP-2571: map cristin contributors to nva-contributors

### PR DESCRIPTION
**Pull request by axthosarouris**
Sorry for the big PR. 

Much of the code is basically new fields reflecting the data in the Json files and generating the necessary testing data.

The JIRA task description:
Map: name, role, sequence,  set corresponding=false if corresponding is not found in Cristin

Later: Id. Put CristinID at the end of URI instead of ARPId
```
{
    "type" : "Contributor",
    "identity" : {
         "type" : "Identity",
          "id" : "https://api.test.nva.aws.unit.no/person/90390302",
           "name" : "Øiseth, Ole Vidar"
     },
     "role" : "Creator",
      "sequence" : 3,
      "correspondingAuthor" : false
}
```

### CODE REVIEW

File: cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinMapper.java [hunk 1]
```diff
@@ -130,6 +142,6 @@ private URI extractLanguage() {
     }
 
     private AdditionalIdentifier extractIdentifier() {
-        return new AdditionalIdentifier(CristinObject.IDENTIFIER_ORIGIN, cristinObject.getId());
+        return new AdditionalIdentifier(CristinObject.IDENTIFIER_ORIGIN, cristinObject.getId().toString());
```
Comment (axthosarouris) [inline cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinMapper.java:145]: id is an Integer

File: cristin-import/src/test/java/no/unit/nva/cristin/AbstractCristinImportTest.java [hunk 2]
```diff
@@ -12,18 +12,31 @@
...
         objectMapperWithEmpty.getTypeFactory().constructCollectionType(List.class, CristinObject.class);
-    protected String testingData;
-
+    public static final String TESTING_DATA_INITIALIZATION_ERROR =
+        "Set the field testingData before calling this method";
+    public static final String INVALID_DATA_ERROR_MESSAGE = "The 'testingData' field does not contain valid data";
 
+    protected String testingData;
 
+    /**
```
Comment (axthosarouris) [inline cristin-import/src/test/java/no/unit/nva/cristin/AbstractCristinImportTest.java:29]: Added some documentation about how the data are fed in the tests

File: cristin-import/build.gradle [hunk 3]
```diff
@@ -61,6 +61,7 @@ dependencies {
 
 test {
     environment "MAX_SLEEP_TIME", "1"
+    environment "CREATE_CONTRIBUTOR_ID", "true"
```
Comment (axthosarouris) [inline cristin-import/build.gradle:64]: Terje wants to not include any id for now but I need it for testing.

File: cristin-import/src/test/java/no/unit/nva/cristin/mapper/ContributionReference.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
+package no.unit.nva.cristin.mapper;
```
Comment (axthosarouris) [inline cristin-import/src/test/java/no/unit/nva/cristin/mapper/ContributionReference.java:1]: This is a class in tests

File: cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinContributorsAffiliation.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    @JsonProperty("undavdnr")
+    private Integer subdepartmentIdentifier;
+    @JsonProperty("gruppenr")
+    private Integer groupNumber;
+    @JsonProperty("stedkode_opprinnelig")
+    private String originalInsitutionCode;
+    @JsonProperty("institusjonsnavn_opprinnelig")
+    private String originalInstitutionName;
+    @JsonProperty("avdelingsnavn_opprinnelig")
+    private String oringalDepartmentName;
```
Comment (brinxmat) [inline cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinContributorsAffiliation.java:34]: Typo

**Review by brinxmat — DISMISSED** [review]

**Pull request merged**